### PR TITLE
website: example of python function app

### PR DESCRIPTION
**Pull request by jfreeland**
I apologize in advance.  It might be that this PR does not make sense to include and is otherwise irrelevant.

I was recently exploring `azurerm_function_app`'s and spent time troubleshooting as documented in https://github.com/Azure/azure-functions-python-worker/issues/627 what was a simple miss of including `FUNCTIONS_WORKER_RUNTIME`.  While the term is mentioned in this page, having a more complete example in this page might have saved me some time; maybe it can also help others.

### CODE REVIEW

File: website/docs/r/function_app.html.markdown [hunk 1]
```diff
@@ -128,6 +128,63 @@ resource "azurerm_function_app" "example" {
 }
 ```
 ~> **Note:** Version `~3` is required for Linux Function Apps.
+~> **Note:** Linux Function Apps will be available separately as `azurerm_linux_function_app` **as an opt-in Beta** in [the upcoming version 3.0 of the Azure Provider](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/3.0-overview).
```
Comment (jackofallops) [inline website/docs/r/function_app.html.markdown:131]: There's a deprecation notice coming for this resource as it's being superseded.
```suggestion
```

File: website/docs/r/function_app.html.markdown [hunk 2]
```diff
@@ -128,6 +128,63 @@ resource "azurerm_function_app" "example" {
...
+
+## Example Usage (Python in a Consumption Plan)
+
+```hcl
+resource "azurerm_resource_group" "example" {
+  name     = "azure-functions-cptest-rg"
```
Comment (jackofallops) [inline website/docs/r/function_app.html.markdown:137]: The convention in these examples is to use `example` over `test`
```suggestion
  name     = "azure-functions-example-rg"
```

File: website/docs/r/function_app.html.markdown [hunk 3]
```diff
@@ -128,6 +128,63 @@ resource "azurerm_function_app" "example" {
...
+  location = "West Europe"
+}
+
+resource "azurerm_storage_account" "example" {
+  name                     = "functionsapptestsa"
```
Comment (jackofallops) [inline website/docs/r/function_app.html.markdown:142]: as above
```suggestion
  name                     = "functionsappexamlpesa"
```

File: website/docs/r/function_app.html.markdown [hunk 4]
```diff
@@ -128,6 +128,63 @@ resource "azurerm_function_app" "example" {
...
+  resource_group_name      = azurerm_resource_group.example.name
+  location                 = azurerm_resource_group.example.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_app_service_plan" "example" {
+  name                = "azure-functions-test-service-plan"
```
Comment (jackofallops) [inline website/docs/r/function_app.html.markdown:150]: as above
```suggestion
  name                = "azure-functions-example-sp"
```

File: website/docs/r/function_app.html.markdown [hunk 5]
```diff
@@ -128,6 +128,63 @@ resource "azurerm_function_app" "example" {
...
+
+  lifecycle {
+    ignore_changes = [
+      kind
+    ]
+  }
+}
+
+resource "azurerm_function_app" "example" {
+  name                       = "test-azure-functions"
```
Comment (jackofallops) [inline website/docs/r/function_app.html.markdown:169]: as above
```suggestion
  name                       = "example-azure-function"
```

File: website/docs/r/function_app.html.markdown [hunk 6]
```diff
@@ -128,6 +128,63 @@ resource "azurerm_function_app" "example" {
...
+  app_settings {
+    FUNCTIONS_WORKER_RUNTIME = "python"
+  }
+
+  site_config {
+    linux_fx_version = "python|3.9"
+  }
+}
+```
+~> **Note:** While you can develop your Python based Azure Functions locally on Windows, Python is only supported on a Linux based hosting plan when running in Azure.  See [the documentation for additional information](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-python).
```
Comment (jackofallops) [inline website/docs/r/function_app.html.markdown:187]:
```suggestion
~> **Note:** The Python runtime is only supported on a Linux based hosting plan.  See [the documentation for additional information](https://docs.microsoft.com/en-us/azure/azure-functions/functions-reference-python).
```

**Review by jackofallops — CHANGES_REQUESTED** [review]
Hi @jfreeland - Thanks for submitting this example for the docs. There's just a few changes in-line below.

It's worth mentioning that this resource is going to be deprecated as part of the 3.0 release, and is being superseded by [azurerm_windows_function_app](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/windows_function_app) and [azurerm_linux_function_app](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/linux_function_app) which both have significant improvements around the handling of the application stacks.

Thanks again!

**Review by katbyte — APPROVED** [review]
Thanks @jfreeland - LGTM 🦀

**Pull request merged**